azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/resources/template/aca/artifact-dockerfile

```diff
@@ -0,0 +1,5 @@
+ARG JAVA_VERSION=17
```

Comment:
what if user's project is developped upon Java 21?

Reply:
It will use Java 21 and 17 is a default version if no java version detected.
Here we detect the java version and add it into buildImageConfig.sourceBuildEnv (we just reuse this map). [code link](https://github.com/microsoft/azure-maven-plugins/blob/79872af2e2c002f0d63d5f52fc61a763df6d2050/azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/parser/ConfigParser.java#L77)
And then we put this into ACR build args [code link](https://github.com/microsoft/azure-maven-plugins/blob/79872af2e2c002f0d63d5f52fc61a763df6d2050/azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/containerapp/ContainerAppDraft.java#L281)
These build args will override the JAVA_VERSION=17 in the dockerfile.